File: frontend/src/types/product.ts
export interface Product {
  id: number;
  name: string;
  description: string;
  cover_image: string | null;
  demo_video_link: string | null;
  frontend_url: string;
  tiers?: Array<Tier>;
  category?: string;
}

export interface ProductCreate {
  name: string;
  description: string;
  cover_image?: string;
  demo_video_link?: string;
  frontend_url: string;
}

export interface ProductUpdate {
  name?: string;
  description?: string;
  cover_image?: string;
  demo_video_link?: string;
  frontend_url?: string;
}

File: app/schemas/product.py
from pydantic import BaseModel
from typing import Optional, List

class ProductBase(BaseModel):
    name: str
    description: str
    cover_image: Optional[str] = None
    demo_video_link: Optional[str] = None
    frontend_url: str

class ProductCreate(ProductBase):
    pass

class ProductUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    cover_image: Optional[str] = None
    demo_video_link: Optional[str] = None
    frontend_url: Optional[str] = None

class Product(ProductBase):
    id: int

    class Config:
        from_attributes = True

class TierBase(BaseModel):
    name: str
    description: str
    price: float
    billing_period: str
    tokens: int
    stripe_price_id: str

class TierCreate(TierBase):
    product_ids: List[int]  # List of product IDs

class Tier(TierBase):
    id: int
    products: List[Product] = []

    class Config:
        from_attributes = True
